addons/account/models/account_reconcile_model.py

```diff
@@ -863,6 +863,11 @@ def _get_invoice_matching_rule_result(self, st_line, candidates, aml_ids_to_excl
             # A write-off must be applied if there are some 'new' lines to propose.
             write_off_lines_vals = list(filter(lambda x: 'id' not in x, lines_vals_list))
             if not lines_vals_list or write_off_lines_vals:
+                for line_vals in write_off_lines_vals:
+                    if line_vals.get('currency_id') and line_vals.get('currency_id') != self.env.company.currency_id.id:
+                        currency = self.env['res.currency'].browse(line_vals['currency_id'])
+                        for key in ('debit', 'credit', 'balance'):
+                            line_vals[key] = self.env.company.currency_id._convert(line_vals[key], currency, self.env.company, fields.Date.today(), False)
```

Comment:
I think today isn't the right date for that.
Instead, I think the right date is the date of the bank statement.